src/channels.rs

```diff
@@ -24,20 +25,33 @@ impl MiEI {
         Ok(())
     }
 
-    pub fn get_role_id(&self, role_name: &str) -> Option<RoleId> {
-        self.courses
-            .values()
-            .filter_map(|x| x.courses.get(role_name))
-            .next()
-            .map(|x| x.role.parse::<RoleId>().unwrap())
+    pub fn get_role_id(&self, role_name: &str) -> Vec<RoleId> {
+        let years = &self.courses;
+        let regex = Regex::new("([0-9]+)ANO([0-9]+)(SEMESTRE|SEM)").unwrap();
+        let year_regex = Regex::new("([0-9])+ANO").unwrap();
```

Comment:
Please use a `lazy_static`: https://docs.rs/regex/1.2.1/regex/#example-avoid-compiling-the-same-regex-in-a-loop